pkg/operator/operator_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/certificates/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+// Mock asynchronous kube-apiserver.
+func mockKubeAPIServer(ctx context.Context, client *fake.Clientset, t *testing.T) {
+	var (
+		csr       *v1.CertificateSigningRequest
+		timeout   = 3 * time.Second
+		fqdn      = fmt.Sprintf("system:node:%s.%s.svc", NameOperator, "")
```

Comment:
nit - can you put in `"test-ns"` or something over `""` for the namespace to make clear?

Reply:
That causes this error `panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1c0 pc=0x172323a]`
@fabxc any tips?

Reply:
resolved.